keps/sig-windows/3746-windows-root-fs-volume-size/README.md

```diff
@@ -0,0 +1,884 @@
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+Windows containers are created with a default root-fs volume size of 20Gb.
+This is sufficient for many workloads and many workloads can use emptyDir or persistent storage volumes with more capacity but there some use cases where it is desirable to have a larger root-fs volume.
+Container runtimes have ways of specifying a non-default root-fs size per container this functionality should be exposed in Kubernetes.
+
+Docker allows users to specify scratch/root-fs size with [**storage-opts**](https://learn.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/container-storage#storage-limits) and CRI-API has [WindowsContainerResources.rootfs-size-in-bytes](https://github.com/kubernetes/kubernetes/blob/ad18954259eae3db51bac2274ed4ca7304b923c4/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto#L995-L996) which is honored by containerd.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+Allow users to specify the size of root-fs volumes for Windows containers in Pod specs.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Allow `resources.limits.storage` to be specified on Containers in pod specs (only if Pod OS is set ot Windows??) and pass the value to container runtime `CreateContainer` calls by setting the existing `WindowsContainerResources.rootfs-size-in-bytes` field in the CRI-API.
+
+Ex:
+
+```yaml
+kind: Pod
+metadata:
+  name: some-pod
+spec:
+  containers:
+  - name: container1
+    resources:
+      limits:
+        storage: 30Gi
```

Comment:
agree `storage` is ambiguous in how it interacts with mounted volumes... would a more explicit `rootfs-storage` be better?

also, is there a reason this would not apply to both linux and windows?

Reply:
what was the resolution of the name `storage` being ambiguous?